### PR TITLE
Add/_merge_with_defaults

### PR DESCRIPTION
**Pull request by rgadon107**
@hellofromtonya PR#12 is ready for code review. Testing the function '_merge_with_defaults()' in the file: 

`/wp-content/mu-plugins/central-hub/src/config-store/internals.php`.

### CODE REVIEW

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_mergeWithDefaults.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	 *
+	 */
+	public function test_should_merge_config_array_with_defaults() {
+		$config = (array) require CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		$defaults = [];
+		$actual = _merge_with_defaults( (array) $config, (array) $defaults );
+
+		$this->assertEmpty( $defaults );
+		$this->assertArrayHasKey( 'foo', $actual );
+		$this->assertSame( $config, $actual );
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/_mergeWithDefaults.php]: @rgadon107 The test is not that the defaults are empty.  Rather, you are wanting to test the following edge cases:

1. If the defaults are empty, then there's not change to the `$config` array.
2. If you have defaults, then test that the `$config` parameters merged with the `$default` parameters and returned exactly what you'd expect.

So let's redo this test and split up into 2 separate tests:

1. `test_should_return_config_when_no_defaults`
2. `test_should_merge_config_parameters_with_defaults`

```php
/**
 *  Test should return the original configuration when no defaults are given.
 */
public function test_should_return_config_when_no_defaults() {
     $config = [
          'foo' => [
		'aaa' => 'bbb',
		'ccc' => 'ddd',
          ]
     ];
     $defaults = [];

     $actual = _merge_with_defaults( $config, $defaults );
     $this->assertSame( $config, $actual );
}

/**
 *  Test should return merge the configuration and default parameters.
 */
public function test_should_merge_config_parameters_with_defaults() {
     $config = [
          'foo' => [
		'aaa' => 'bbb',
		'ccc' => 'ddd',
          ]
     ];
     $defaults = [
          'foo' => [
		'aaa' => 1,
		'ccc' => 2,
		'ddd' => 3,
          ]
     ];

     $actual = _merge_with_defaults( $config, $defaults );

     $expected = [
          'foo' => [
		'aaa' => 'bbb',
		'ccc' => 'ddd',
		'ddd' => 3,
          ]
     ];
     $this->assertSame( $expected, $actual );
}
```
             


```

Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/_mergeWithDefaults.php]: @hellofromtonya In response to your comments, I updated the branch and revised the tests. [See today's commit.](https://github.com/rgadon107/cornerstone/pull/12/commits/bcd681a8d2112cff37fc5b479d63631c09027b03)
